pyiceberg/io/pyarrow.py

```diff
@@ -362,6 +362,12 @@ def _initialize_fs(self, scheme: str, netloc: Optional[str] = None) -> FileSyste
                 "region": get_first_property_value(self.properties, S3_REGION, AWS_REGION),
             }
 
+            # Override the default s3.region if netloc(bucket) resolves to a different region
+            try:
+                client_kwargs["region"] = resolve_s3_region(netloc)
```

Comment:
What about doing this lookup only when the region is not provided explicitly? I think this will do another call to S3.

Reply:
Thank you Fokko, my understanding is that the problem occurs when the provided `region` doesn't match the data file bucket region, and that will fail the file read for pyarrow.